amethyst_renderer/src/mesh.rs

```diff
@@ -65,7 +65,11 @@ where
 
         let verts = self.0.as_ref();
         let slice = cast_slice(verts);
-        let stride = slice.len() / verts.len();
+        let verts_len = verts.len();
+        if verts_len == 0 {
+            return Err(Error::from_string("Number of vertices in mesh is zero."));
```

Comment:
I think this could be a `log::warn` instead. Hard errors on user controlled data are annoying to deal with. When `verts_len` is 0, `slice.len()` will be 0 as well. The stride can be just set to 0 and it should all work out.

Reply:
Why have the Result when we're not going to use it?

Reply:
Actually, I think the better question for me to ask would be is it useful to actually build empty vertex buffers? You would know this better I think. I see how the Result is being used now, but I'm not sure whether you care about having empty vertex buffers sent to the GPU

Reply:
Well, actually creating 0-sized buffers seems like a problem, and there isn't much use in empty meshes anyway. The draw call with empty mesh, if at all valid, would result in no primitives drawn.

I'm just a bit worried that this might cause some spontaneous crashes on hot-reloading when somebody will clear up the mesh in their's editor and save the file. We might handle that in asset pipeline though :)

Reply:
Gotcha. Gonna use the code suggested by @vessd and instead warn!+return zero in the closure.